include/libdnf-cli/output/environmentinfo.hpp

```diff
@@ -109,6 +109,7 @@ template <typename EnvironmentType>
 void print_environmentinfo_table(EnvironmentType & environment) {
     struct libscols_table * table = create_environmentinfo_table(environment);
     scols_print_table(table);
+    scols_unref_symbols(scols_table_get_symbols(table));
```

Comment:
No. You are fixing the memory leak in the wrong place. Memory leak occurs earlier in the `create_environmentinfo_table` function.

Symbols use reference counting. The `create_environmentinfo_table` function does:
1. `auto sy = scols_new_symbols();` - allocates new symbols with reference counter set to 1
2. `scols_table_set_symbols(table, sy);` - add reference to `sy` from the table, symbols reference counter is incremented to 2
3. The variable `sy` is not longer used and there is a memory leak, please add the line `scols_unref_symbols(sy);` immediately after `scols_table_set_symbols(table, sy);` to decrement the reference counter back to 1.

And then `scols_unref_table(table);` will properly decrement the reference counter to 0 and memory will be freed.

Please fix it on other places as well.

Reply:
I see, fixed.